pkg/controller/windowsmachine/windows/windows.go

```diff
@@ -161,31 +165,39 @@ func (vm *windows) Configure() error {
 }
 
 func (vm *windows) ConfigureHybridOverlay(nodeName string) error {
-	// Check if the hybrid-overlay is running
-	_, err := vm.Run("Get-Process -Name \""+HybridOverlayProcess+"\"", true)
-
-	// err being nil implies that hybrid-overlay is running.
-	if err == nil {
-		// Stop the hybrid-overlay
-		stopCmd := "Stop-Process -Name \"" + HybridOverlayProcess + "\""
-		out, err := vm.Run(stopCmd, true)
-		if err != nil {
-			log.Info("unable to stop hybrid-overlay", "stop command", stopCmd, "output", out)
-			return errors.Wrap(err, "unable to stop hybrid-overlay")
-		}
-	}
 	var customVxlanPortArg = ""
 	if len(vm.vxlanPort) > 0 {
 		customVxlanPortArg = " --hybrid-overlay-vxlan-port=" + vm.vxlanPort
 	}
-	// Start the hybrid-overlay in the background over ssh.
-	go vm.Run(k8sDir+wkl.HybridOverlayName+" --node "+nodeName+customVxlanPortArg+
-		" --k8s-kubeconfig c:\\k\\kubeconfig --logfile="+hybridOverlayLogDir+"hybrid-overlay.log", false)
 
-	if err = vm.waitForHybridOverlayToRun(); err != nil {
-		return errors.Wrapf(err, "error running %s", wkl.HybridOverlayName)
+	hybridOverlayServiceArgs := "--node " + nodeName + customVxlanPortArg + " --k8s-kubeconfig c:\\k\\kubeconfig " +
+		"--windows-service " + "--logfile " + hybridOverlayLogDir + "hybrid-overlay.log\" depend= " + kubeletServiceName
+
+	hybridOverlayService, err := newService(hybridOverlayPath, hybridOverlayServiceName, hybridOverlayServiceArgs)
+	if err != nil {
+		return errors.Wrapf(err, "error creating service object for %s", hybridOverlayServiceName)
 	}
 
+	// if the service exits and it is running, then stop it.
+	if vm.serviceExists(hybridOverlayServiceName) {
+		if vm.isRunning(hybridOverlayServiceName) {
+			if err := vm.stopService(hybridOverlayService); err != nil {
+				return errors.Wrapf(err, "error stopping %s Windows service", hybridOverlayServiceName)
+			}
+		}
+	} else {
+		if err := vm.createService(hybridOverlayService); err != nil {
```

Comment:
Can't this be combined? If yes, please do that for the kube-proxy case too.

Reply:
Please note the comment is about two lines, 188 to 189. I am asking if this can be made into an `if else`. But thinking about your question over slack, I think lines 182-192 can be made into a function like `ensureServiceExists()` and used for both kube-proxy and hybrid-overlay services.